plugins/feeds/public/virustotal_apiv3.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import re
+import json
+from datetime import timedelta, datetime
+
+from core import Feed
+from core.config.config import yeti_config
+from core.observables import Hash, File
+
+# Variable
+VTAPI = yeti_config.get('vt', 'key')
+headers = {"x-apikey": VTAPI}
+limit = 10
+params = {'limit': limit}
+regex = "[A-Fa-f0-9]{64}"  # Find SHA256
+
+
+class VirusTotalPriv(Feed):
+    default_values = {
+        "frequency": timedelta(minutes=5),
+        "name": "VirusTotalHuntingV3",
+        "source": "https://www.virustotal.com/api/v3/intelligence/hunting_notifications",
+        "description": "Feed of hunting for VirusTotal API v3",
+    }
+
+    settings = {
+            'vt_url_hunting_v3': {
+            'name': 'VT Url Hunting v3',
+            'description': 'Hunting feed for VT API v3'
+        }
+    }
+
+    def update(self):
+        if VTAPI:
+            self.source = "https://www.virustotal.com/api/v3/intelligence/hunting_notifications"
+            for index, item in self.update_json(params=params, headers=headers, key="data"):
+                self.analyze(item)
+        else:
+            logging.error("Your VT API key is not set in the config file!")
+
+    def analyze(self, item):
+        tags = []
+
+        # Convert data to json
+        json_string = item.to_json()
+        json_string = json.loads(json_string)
+
+        context = {'source': self.name}
+
+        # Parse value of interest
+        subject = json_string["attributes"]["rule_name"]
+        date = json_string["attributes"]["date"]
+        tags2 = json_string["attributes"]["tags"]
+        sha2 = re.search(regex, str(tags2)).group()
```

Comment:
why use a regex to have the hash ? it's not in present in the data ?

Reply:
The new format of the request with the VT APIv3 doesn't provide a key with the hash but rather put the hash into the tags key. Thus there is a need to identify which tags is the Sha2.